File: Developer/AppFrameworks/YJFoundation/YJFoundationTests/YJURLSessionTest/YJTestURLSessionTask.m
//
//  YJTestURLSessionTask.m
//  YJFoundation
//
//  Created by 阳君 on 2016/11/30.
//  Copyright © 2016年 YJCocoa. All rights reserved.
//

#import "YJTestURLSessionTask.h"

@implementation YJTestURLSessionTask

- (void)resume {
    [super resume];
    self.failure([NSError errorWithDomain:@"网络错误测试" code:404 userInfo:nil]);
}

@end

File: Demo/OC/AppFrameworks/Foundation/YJFoundation/YJFoundationTests/YJTestURLSessionTask.swift
//
//  YJTestURLSessionTask.swift
//  YJFoundationTests
//
//  Created by 阳君 on 2019/6/21.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import UIKit
import YJCocoa

@objcMembers
class YJTestURLRequestModel: NSObject {
    var name: String = ""
    var qq: String = ""
}

@objcMembers
class YJTestURLResponseModel: NSObject {
    var desc: String?
}

class YJTestURLSessionTask: YJURLSessionTask {
    override func resume() {
        guard self.state != .running else {
            return
        }
        super.resume()
        let json = "{\"desc\": \"服务器回调\"}"
        self.success(json)
    }
}
